src/pages/ReimbursementAccount/ReimbursementAccountPage.js

```diff
@@ -97,116 +97,175 @@ const defaultProps = {
-        const hasACHDataBeenLoaded = this.props.reimbursementAccount !== ReimbursementAccountProps.reimbursementAccountDefaultProps;
-        this.state = {
-            hasACHDataBeenLoaded,
-            shouldShowContinueSetupButton: hasACHDataBeenLoaded ? this.getShouldShowContinueSetupButtonInitialValue() : false,
-        };
-    }
-
-    componentDidMount() {
-        this.fetchData();
-    }
-
-    componentDidUpdate(prevProps) {
-        if (prevProps.network.isOffline && !this.props.network.isOffline && prevProps.reimbursementAccount.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) {
-            this.fetchData();
+function ReimbursementAccountPage({
+    reimbursementAccount,
+    network,
+    route,
+    onfidoToken,
+    policy,
+    account,
+    isLoadingReportData,
+    session,
+    translate,
+    plaidLinkToken,
+    reimbursementAccountDraft,
+
+}) {
+    const [prevProps, setPrevProps] = useState({
+        current: {
+            reimbursementAccount,
+            network,
+            route,
+            onfidoToken,
+            policy,
+            account,
+            isLoadingReportData,
+            session,
+            translate,
+            plaidLinkToken,
+            reimbursementAccountDraft,
         }
-        if (!this.state.hasACHDataBeenLoaded) {
-            // If the ACHData has not been loaded yet, and we are seeing the default data for props.reimbursementAccount
-            // We don't need to do anything yet
-            if (this.props.reimbursementAccount !== ReimbursementAccountProps.reimbursementAccountDefaultProps && !this.props.reimbursementAccount.isLoading) {
-                // If we are here, it is because this is the first time we load the ACHData from the server and
-                // this.props.reimbursementAccount.isLoading just changed to false. From now on, it makes sense to run the code
-                // below updating states and the route, and this will happen in the next react lifecycle.
-                this.setState({
-                    shouldShowContinueSetupButton: this.getShouldShowContinueSetupButtonInitialValue(),
-                    hasACHDataBeenLoaded: true,
-                });
+    });
+    const [shouldShowContinueSetupButton, setShouldShowContinueSetupButton] = useState(
+        hasACHDataBeenLoaded ? getShouldShowContinueSetupButtonInitialValue() : false
+    );
+    const [hasACHDataBeenLoaded, setHasACHDataBeenLoaded] = useState(reimbursementAccount !== ReimbursementAccountProps.reimbursementAccountDefaultProps);
+
+    const achData = lodashGet(reimbursementAccount, 'achData', {});
+    const currentStep = achData.currentStep || CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT;
+    const policyName = lodashGet(policy, 'name');
+    const policyID = lodashGet(route.params, 'policyID');
+
+    const continueFunction = () => {
+        setShouldShowContinueSetupButton(false);
+        fetchData(true);
+    };
+
+    useEffect(() => {
```

Comment:
You should do it within `memo` hoc, not inside a component.

Reply:
As we discussed I removed the prevProps totally. Finally I used `useRef` to deal with the update. Please check if its ok.